internal/utils.go

```diff
@@ -326,23 +329,31 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]interface{
 // When index is 0, it won't attach any number. When useManifestName is true,
 // it returns the policyConf.Manifests[manifestGroupIndex].Name otherwise,
 // It returns policyConf.Name + index
-func getConfigurationPolicyName(policyConf *types.PolicyConfig, manifestGroupIndex, policyNum,
-	manifestIndex int, useManifestName bool,
+func getConfigurationPolicyName(policyConf *types.PolicyConfig, manifestGroupIndex int,
+	usedPolicyNameNum *int, usedManifestNameNum *int,
 ) string {
 	// Do not append a number to configName when it is used for the first time.
 	configName := policyConf.Manifests[manifestGroupIndex].Name
 
-	if useManifestName && configName != "" {
-		if manifestIndex > 0 {
-			return fmt.Sprintf("%s%d", configName, manifestIndex+1)
+	// For the case where ConsolidateManifests is true and the manifest's name is provided,
+	if usedManifestNameNum == nil && configName != "" {
+		return configName
+	}
+
+	if usedManifestNameNum != nil && configName != "" {
+		*usedManifestNameNum++
+		if *usedManifestNameNum > 1 {
+			return fmt.Sprintf("%s%d", configName, *usedManifestNameNum)
 		}
 
 		return configName
 	}
```

Comment:
Should we add error message or warning message for this, otherwise this is really confusing usesr.

Reply:
Brian commented https://issues.redhat.com/browse/ACM-12563

Reply:
I don't think it's confusing for the user since the manifests are being combined so it's using the parent configuration, and I _think_ (?) this PR resolves Brian's code bug since the `manifest[].name` is handled separately for `object-templates-raw`.

Reply:
Oh, I see. I wasn't thinking about this allowing configuration of the ConfigurationPolicy to have a different name than the Policy. Yes, this makes sense then to just use the first name it finds.

Reply:
So leave it as t is?

Reply:
Leave the logic, yes. I still think the conditionals could optionally be combined:
```golang
// The manifest name is provided
if configName != "" {
	// The manifest name counter is not nil so must be incremented and used if it's greater than one
	if usedManifestNameNum != nil {
		*usedManifestNameNum++
		if *usedManifestNameNum > 1 {
			return fmt.Sprintf("%s%d", configName, *usedManifestNameNum)
		}
	}

	// Return the manifest name without a suffix
	return configName
}
```
I might have to think more about whether we should be doing duplicate name validation, but right now I'm thinking that can be chalked up to user error since the framework would complain once it's deployed.

Reply:
I am not sure this repos has sonarcloud. Sonar doesn't like too many if-if child.